src/relentless/optimize/method.py

```diff
@@ -54,6 +56,21 @@ def stop(self, value):
             raise TypeError("The stopping criterion must be a ConvergenceTest.")
         self._stop = value
 
+    def setup_directory(self, directory=None, overwrite=False):
```

Comment:
Let's make this private since we don't want the user to call it. Also, let's require the user to specify both arguments:
```suggestion
    def _setup_directory(self, directory, overwrite):
```